.github/workflows/native-unix.yml

```diff
@@ -332,7 +332,7 @@ jobs:
           sudo sysctl vm.mmap_rnd_bits=28
 
       - name: Install staticcheck
-        run: go install honnef.co/go/tools/cmd/staticcheck@v0.3.3
+        run: go install honnef.co/go/tools/cmd/staticcheck@latest
       - name: Go Build
```

Comment:
Hmm, this and below need an `if: matrix.os == 'ubuntu-latest'`

Reply:
Why? Wouldn't we want to still run the staticcheck linting on macos too?

Reply:
It doesn't work on macOS, and we've always excluded it (I'm not sure how this passed before actually)

Reply:
https://github.com/apache/arrow-adbc/actions/runs/8349298201/job/22853175768

Reply:
but IDK, this is the commit where that check came from: https://github.com/apache/arrow-adbc/commit/fb09decd8e38fc4d3e191a8d2d186568e2640d69

Reply:
That failure was being caused by the old pinned version of staticcheck. when i bumped the staticcheck version that problem went away